README.md

```diff
@@ -0,0 +1,68 @@
+# bdcat_notebooks
+Example Jupyter notebook source files for the BioData Catalyst consortium. These notebooks are published to the
+[Terra notebook environment](https://support.terra.bio/hc/en-us/articles/360027237871-Terra-s-Jupyter-Notebooks-environment-Part-I-Key-components).
+
+Jupyter notebook `.ipynb` files are generated from source files with the
+[callysto](https://github.com/xbrianh/callysto) library. 
+
+## Source File Schema
+The first line of each notebook source file contains the publish directive, a string with the following format:
+```
+# publish to "{my workspace name}" "{my notebook name}"
+```
+where `{my workspace name}` should be replaced with the name of the workspace, and `{my notebook name}` should be
+replaced with the name of the notebook.
+
+Notebook source files are executable Python scripts. The contents of cells are denoted with the callysto Python
+context managers:
+```
+with callysto.Cell("{mode}"):
+	...
+```
+where `{mode}` is `python` or `markdown`. Python statements outside if callysto contexts will not be rendered into
+cells. Test code, or code for preparing mock environments, should not be placed in callysto contexts.
+
+## Publishing Notebooks
+Notebooks are published with make commands, e.g.
+```
+make notebooks/prepare_igv_viewer_input.py
+```
+
+This command passes the source script through the [flake8](https://flake8.pycqa.org/en/latest/) linter and
+[mypy](https://mypy.readthedocs.io/en/stable/) static analysis tool, and executes with the local Python interpreter.
+If there are no errors, callysto is used to generate the source script into an `.ipynb`, which is copied into the Terra
+workspace bucket.
+
+Publishing requires Google client authorization, as well as Terra access to the target workspace. Google authorization
+is obtained with the command
+```
+gcloud auth login
+```
```

Comment:
This might not need to be in the docs but I'm curious -- how does this handle the fact UCSC Google login now requires 2FA?

Reply:
2FA is handled entirely by `gcloud`. It shouldn't be a problem, and in any case is not related to the contents of this repo.